ADR/0033-upstreaming-integration-service.md

```diff
@@ -0,0 +1,228 @@
+
+[integration-service] should grow a mutating webhook that *removes* the `CONTROLLABEL` label from
+the last control [TestSubject] for the `LABELNAME` label, when the `CONTROLLABEL` label is applied
+to a new [TestSubject] bearing the `LABELNAME` label.
+
+### TestSubject Construction as optional behavior
+
+**Test Initiation** has a clear control plane: the [IntegrationTestScenario] combined with the
+[TestSubject]. Meaning, when the user creates an [IntegrationTestScenario] and a [TestSubject], and
+the [selector] of the [IntegrationTestScenario] matches the attributes of [TestSubject], then together
+they represent intent for the subject to be tested. The user uses these resources to control what
+the system does.
+
+Today, control of the **Snapshot Construction** process is mixed up in the [Application] and [Component]
+model, and is partly just implied. We *always* construct [Snapshots].
+
+**TestSubject Construction** (called Snapshot Construction in our system today) should have a clear
+control plane resource: the [ToBeDetermined] resource.
+
+TODO - describe this in more detail
+
+TODO - describe the spec of this resource.
+
+### Release Initiation, as owned by the release-service
+
+Note, in this pattern, [Release] construction looks a lot like [TestSubject] construction. The
+triggered system knows enough about the triggering system to optionally initiate itself.
+
+TODO - describe this in more detail
+
+> :memo: Another alternative could be to establish an overarching workflow operator that knows how
+> to bridge the gaps (create TestSubjects and create Releases). TODO what would this look like? Too
+> much domain knowledge in one place?
+
+## Use Cases in Detail
+
+**Using integration-service as a remote API from Jenkins.** A hypothetical platform team might want
+to make use of [integration-service] from a jenkins pipeline. For example, perhaps their jenkins job
+performs CI on a git repository that contains pullspecs which should be mirrored to some environment
+of theirs. Before mirroring, they want to make sure that the same test works against dozens of
+different kubernetes configurations on different public clouds. That platform engineering team first
+creates a single tekton pipeline that contains the logic to perform the test given a single cluster.
+They then configure dozens of [IntegrationTestScenario] resources, one for each kubernetes
+configuration they want to target. When a new set of pullspecs is proposed to their git repo,
+Jenkins responds. Jenkins then creates a single [TestSubject] in the kubernetes namespace,
+representing the new set of images under test. [integration-service] responds, inspecting all of the
+defined [IntegrationTestScenarios], it invokes the [deployment-target-operator] (from [Decoupling
+Deployment]) to provision each cluster configuration and it runs tests of the [TestSubject] against
+each. The aggregate results and stored in the `status` of the [TestSubject] for jenkins to monitor
```

Comment:
Would it make sense to enable a flow to only re-run specific IntegrationTestScenarios? If during the creation of a TestSubject, the user knows which specific ITSs it wants to run, it seems reasonable that only those desired test scenarios could be triggered. 

This does, however, complicate the decision around moving the `CONTROLLABEL` as all required ITSs configured would need to pass, not just those that were triggered by the TestSubject. This could be achieved by copying the status of the ITSs from the most recent TestSubject which shares a common set of artifact references (which is not discussed in this ADR).